spec/truffle/launcher_spec.rb

```diff
@@ -56,58 +56,63 @@ def should_print_full_java_command(options, env: {})
-    should_print_full_java_command "-J-cmd --version"
-  end
+  unless Truffle.native?
+    it "prints the full java command with -J-cmd" do
+      should_print_full_java_command "-J-cmd --version"
+    end
 
-  it "prints the full java command with --jvm.cmd" do
-    should_print_full_java_command "--jvm.cmd --version"
-  end
+    it "prints the full java command with --jvm.cmd" do
+      should_print_full_java_command "--jvm.cmd --version"
+    end
 
-  it "prints the full java command with -cmd in JAVA_OPTS" do
-    should_print_full_java_command "--version", env: { "JAVA_OPTS" => "-cmd" }
-  end
+    it "prints the full java command with -cmd in JAVA_OPTS" do
+      should_print_full_java_command "--version", env: { "JAVA_OPTS" => "-cmd" }
+    end
 
-  it "adds options from $JAVA_OPTS to the command" do
-    option = '-Dfoo.bar=baz'
-    out = ruby_exe(nil, options: "-J-cmd --version", env: { "JAVA_OPTS" => option })
-    parts = out.lines[0].split(' ')
-    parts.find { |part| part =~ /^#{option}$/ }.should_not be_nil
-    $?.success?.should == true
+    it "adds options from $JAVA_OPTS to the command" do
+      option = '-Dfoo.bar=baz'
+      out = ruby_exe(nil, options: "-J-cmd --version", env: { "JAVA_OPTS" => option })
+      parts = out.lines[0].split(' ')
+      parts.find { |part| part =~ /^#{option}$/ }.should_not be_nil
+      $?.success?.should == true
+    end
+
+    it "takes options from system properties set in JAVA_OPTS" do
+      out = ruby_exe("puts $VERBOSE", env: { "JAVA_OPTS" => "-Dpolyglot.ruby.verbosity=true" })
+      $?.success?.should == true
+      out.should == "true\n"
+    end
   end
 
+  property_option_prefix = Truffle.native? ? '--native.' : '-J-'
+
   it "preserve spaces in options" do
-    out = ruby_exe("print Truffle::System.get_java_property('foo')", options: '-J-Dfoo="value with spaces"')
+    out = ruby_exe("print Truffle::System.get_java_property('foo')",
+                   options: property_option_prefix + 'Dfoo="value with spaces"')
     $?.success?.should == true
     out.should == "value with spaces"
   end
 
-  it "takes options from TRUFFLERUBYOPT" do
-    out = ruby_exe("puts $VERBOSE", env: { "TRUFFLERUBYOPT" => "-W2" })
-    $?.success?.should == true
-    out.should == "true\n"
-  end
-
-  it "takes options from RUBYOPT" do
-    out = ruby_exe("puts $VERBOSE", env: { "RUBYOPT" => "-W2" })
+  it "takes options from system properties set on the command line using -J" do
+    out = ruby_exe("puts $VERBOSE", options: property_option_prefix + "Dpolyglot.ruby.verbosity=true")
     $?.success?.should == true
     out.should == "true\n"
   end
 
-  it "takes options from system properties set in JAVA_OPTS" do
-    out = ruby_exe("puts $VERBOSE", env: { "JAVA_OPTS" => "-Dpolyglot.ruby.verbosity=true" })
+  it "takes options from system properties set on the command line using --jvm" do
+    out = ruby_exe("puts $VERBOSE", options: property_option_prefix + "Dpolyglot.ruby.verbosity=true")
```

Comment:
This no longer tests `--jvm`.
I think it's better to just have a spec for the native case rather than using `property_option_prefix`, to clarify which option prefix is used.